Support microseconds in the default timestamp format, CASSANDRA-10428

stef1927 opened the pull request:
Some fixes to support microseconds in the timestamp and a new test for it (3.2+ only). Please wait for CASSANDRA-10428 to be committed before merging.

stef1927 commented:
Gentle ping, could you please review, CASSANDRA-10428 is ready for commit and without this patch some tests will fail.

mambocab commented on cqlsh_tests/cqlsh_tests.py:
```diff
@@ -143,8 +143,44 @@ def test_past_and_future_dates(self):
+            insert into testsubsecond (id, subid, value) VALUES (1, '1943-06-19 11:21:01.123+0000', 'abc');
+            insert into testsubsecond (id, subid, value) VALUES (2, '1943-06-19 11:21:01+0000', 'def')""")
+
+        output, err = self.run_cqlsh(node1, "use simple; SELECT * FROM testsubsecond "
```
Will this run with `node1.cqlsh`? If so, please use it; we're trying to get rid of the `run_cqlsh` reimplemented in dtest.

mambocab commented on cqlsh_tests/cqlsh_tests.py:
```diff
@@ -143,8 +143,44 @@ def test_past_and_future_dates(self):
+        self.assertIn("1943-06-19 11:21:01.123000+0000", output)
+        self.assertNotIn("1943-06-19 11:21:01.000000+0000", output)
+
+        output, err = self.run_cqlsh(node1, "use simple; SELECT * FROM testsubsecond "
```
Same here. You can have a `ccm` `Node.run_cqlsh` return its output as a tuple with the `return_output` kwarg.

mambocab commented:
Sorry for blocking merge on the ticket. The tests look good; if you can, please use the `ccm` implementation of `run_cqlsh` rather than the one in dtest, but once that's addressed, feel free to merge. Thanks!

stef1927 commented:
Thank you for the review. Replaced `self.run_cqlsh` with the ccm implementation and squashed + rebased. Waiting for C* ticket to be committed before merging.

mambocab commented:
Awesome, looks great. Thank you!
